app/code/Transoft/Blog/ViewModel/Bloglist.php

```diff
@@ -0,0 +1,105 @@
+<?php
+declare(strict_types=1);
+namespace Transoft\Blog\ViewModel;
+
+use Magento\Catalog\Block\Product\View;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Transoft\Blog\Model\ResourceModel\Post\Collection;
+use Transoft\Blog\Model\ResourceModel\Post\CollectionFactory as PostCollectionFactory;
+
+/**
+ * Class Bloglist
+ *
+ * Blog list view model class
+ */
+class Bloglist implements ArgumentInterface
+{
+    /**
+     * @var PostCollectionFactory
+     */
+    protected $postCollectionFactory;
+
+    /**
+     * ScopeConfigInterface $scopeInterface
+     */
+    private $scopeInterface;
+
+    /**
+     * UrlInterface $urlInterface
+     */
+    private $urlInterface;
+
+    /**
+     * Constructor.
+     *
+     * @param PostCollectionFactory $postCollectionFactory
+     * @param UrlInterface $urlInterface
+     * @param ScopeConfigInterface $scopeInterface
+     */
+    public function __construct(
+        PostCollectionFactory $postCollectionFactory,
+        UrlInterface $urlInterface,
+        ScopeConfigInterface $scopeInterface
+    ) {
+        $this->postCollectionFactory = $postCollectionFactory;
+        $this->urlInterface = $urlInterface;
+        $this->scopeInterface = $scopeInterface;
+    }
+
+    /**
+     * Returns model collection
+     *
+     * @return Collection
+     */
+    public function getPostCollection()
+    {
+        $collection = $this->postCollectionFactory->create();
+        $collection->setOrder('creation_time', 'DESC')->setPageSize(5);
+
+        return $collection;
+    }
+
+    /**
+     * Check if types matching
+     *
+     * @return bool
+     */
+    public function isTypesMatching() : bool
+    {
+        $types = $this->scopeInterface->getValue('catalog/blog/blog_applied_to');
+        $product = ObjectManager::getInstance()->get(View::class);
```

Comment:
Also, for future - all required dependencies should be defined as constructor arguments

Reply:
fixed